pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TableViewTest.java

```diff
@@ -136,6 +142,134 @@ private Set<String> publishMessages(String topic, int count, boolean enableBatch
         return keys;
     }
 
+    @DataProvider(name = "partition")
+    public static Object[][] partition () {
+        return new Object[][] {
+                { 3 }, { 0 }
+        };
+    }
+
+    /**
+     * Case1:
+     * 1. Slow down the rate of reading messages.
+     * 2. Send some messages
+     * 3. Call new `refresh` API, it will wait for reading all the messages completed.
+     * Case2:
+     * 1. No new messages.
+     * 2. Call new `refresh` API, it will be completed immediately.
+     * Case3:
+     * 1. multi-partition topic, p1, p2 has new message, p3 has no new messages.
+     * 2. Call new `refresh` API, it will be completed after read new messages.
+     */
+    @Test(dataProvider = "partition")
+    public void testRefreshAPI(int partition) throws Exception {
+        // 1. Prepare resource.
+        String topic = "persistent://public/default/testRefreshAPI" + RandomUtils.nextLong();
+        if (partition == 0) {
+            admin.topics().createNonPartitionedTopic(topic);
+        } else {
+            admin.topics().createPartitionedTopic(topic, partition);
+        }
+        @Cleanup
+        Consumer<byte[]> consumer = pulsarClient.newConsumer()
+                .topic(topic)
+                .subscriptionName("my-sub")
+                .subscribe();
```

Comment:
```suggestion
```

Should it be removed?

Reply:
Without a durable cursor on this topic, the messages will be acknowledged automatically, right?

Reply:
Not accurately. An "acknowledgment" only takes effect on a durable subscription (i.e. the subscription with a durable cursor). If a topic has no durable cursor, the reader in `TableView` still works as expected. It only affects the topic retention. For a test, it's meaningless to create a durable cursor on the test topic.

And after removing these code, the test will still pass.

Reply:
The read position of the reader is `startMessageId(MessageId.earliest)` and the retention time is long, so it is fine to delete this.